src/Mod/Fem/Gui/DlgSettingsFemGmshImp.cpp

```
@@ -49,14 +50,41 @@ DlgSettingsFemGmshImp::~DlgSettingsFemGmshImp() = default;
 
 void DlgSettingsFemGmshImp::saveSettings()
 {
+
     ui->cb_gmsh_binary_std->onSave();
     ui->fc_gmsh_binary_path->onSave();
+    ui->cb_mesh_file_format->onSave();
+
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath(
+        "User parameter:BaseApp/Preferences/Mod/Fem/Gmsh");
+    std::string meshFormats[] = {".unv", ".vtk", ".inp", ".msh"};
+    int ind = ui->cb_mesh_file_format->currentIndex();
+    hGrp->SetASCII("MeshFileFormat", meshFormats[ind]);
 }
 
 void DlgSettingsFemGmshImp::loadSettings()
 {
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath(
+        "User parameter:BaseApp/Preferences/Mod/Fem/Gmsh");
+    std::string str = hGrp->GetASCII("MeshFileFormat", ".unv");
+    int ind = 0;
+    if (str == ".unv") {
+        ind = 0;
+    }
+    else if (str == ".vtk") {
+        ind = 1;
+    }
+    else if (str == ".inp") {
+        ind = 2;
+    }
+    else if (str == ".msh") {
+        ind = 3;
+    }
```

I think this should be an error instead of silently using the default.